rbx_reflector/src/cli/defaults_place.rs

```diff
@@ -0,0 +1,247 @@
+                instance.add_child(Instance::new("Terrain"));
+            }
+            "Part" => {
+                instance.add_child(Instance::new("Attachment"));
+                instance.add_child(Instance::new("Bone"));
+            }
+            "Humanoid" => {
+                instance.add_child(Instance::new("Animator"));
+            }
+            "MeshPart" => {
+                // Without this special case, Studio will fail to open the resulting file, complaining about "BaseWrap".
+                instance.add_child(Instance::new("BaseWrap"));
+                instance.add_child(Instance::new("WrapLayer"));
+                instance.add_child(Instance::new("WrapTarget"));
+            }
+
+            _ => {}
+        }
+
+        write!(place_contents, "{}", instance).unwrap();
+    }
+
+    writeln!(place_contents, "</roblox>").unwrap();
+
+    fs::write(path, place_contents)?;
+
+    Ok(())
+}
+
+struct Instance<'a> {
+    class_name: &'a str,
+    children: Vec<Instance<'a>>,
+}
+
+impl<'a> Instance<'a> {
+    fn new(class_name: &'a str) -> Self {
+        Self {
+            class_name,
+            children: Vec::new(),
+        }
+    }
+
+    fn add_child(&mut self, child: Instance<'a>) {
+        self.children.push(child);
+    }
+}
+
+impl fmt::Display for Instance<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(
+            f,
+            "<Item class=\"{}\" referent=\"{}\">",
+            self.class_name, self.class_name
+        )?;
+
+        for child in &self.children {
+            write!(f, "{}", child)?;
+        }
+
+        writeln!(f, "</Item>")?;
+
+        Ok(())
+    }
+}
+
+pub struct PluginInjector<'a> {
+    http_server: tiny_http::Server,
+    studio_install: &'a RobloxStudio,
+}
+
+impl<'a> PluginInjector<'a> {
+    pub fn start(studio_install: &'a RobloxStudio) -> anyhow::Result<Self> {
+        let http_server = tiny_http::Server::http("0.0.0.0:22073").unwrap();
+
+        log::info!("Installing Studio plugin");
+
+        let plugin_path = studio_install
+            .plugins_path()
+            .join("RbxDomDefaultsPlacePlugin.lua");
+
+        fs::write(plugin_path, PLUGIN_SOURCE)?;
+
+        Ok(Self {
+            http_server,
+            studio_install,
+        })
+    }
+
+    pub fn wait_for_response(self) -> anyhow::Result<()> {
+        let request = match self.http_server.recv_timeout(Duration::from_secs(30))? {
+            Some(request) => request,
+            None => bail!("Plugin did not send a response within 30 seconds"),
+        };
+
+        request.respond(Response::empty(200))?;
+
+        Ok(())
+    }
+}
+
+impl<'a> Drop for PluginInjector<'a> {
+    fn drop(&mut self) {
+        log::info!("Uninstalling Studio plugin");
+
+        let plugin_path = self
+            .studio_install
+            .plugins_path()
+            .join("RbxDomDefaultsPlacePlugin.lua");
+
+        fs::remove_file(plugin_path).unwrap();
```

Comment:
Panicking in `Drop` implementations isn't a good idea; it'll likely just abort the program because it can't unwind the panic (the `PanicInfo` it creates will get dropped immediately).

It's almost certainly fine to abort if we panic here but I'd rather we just log if it fails:

```suggestion
        if let Err(err) = fs::remove_file(plugin_path) {
            log::error!("Could not remove plugin: {err}");
        }
```